lib/html/pipeline/emoji_filter.rb

```diff
@@ -13,19 +13,25 @@ class Pipeline
     # Context:
     #   :asset_root (required) - base url to link to emoji sprite
     #   :asset_path (optional) - url path to link to emoji sprite. :file_name can be used as a placeholder for the sprite file name. If no asset_path is set "emoji/:file_name" is used.
+    #   :ignored_ancestor_tags (optional) - Tags to stop the emojification. Node has matched ancestor HTML tags will not be emojified. Default to pre, code, and tt tags. Multiple tags please use comma to separate each tag, and no space between tags, e.g. "h1,h2".
+    #   :ignored_pattern (optional) - Content in matching surrounded pattern will not be emojified, e.g. "`".
     class EmojiFilter < Filter
+
+      DEFAULT_IGNORED_ANCESTOR_TAGS = %w(pre code tt).freeze
+
       def call
         doc.search('text()').each do |node|
           content = node.to_html
           next unless content.include?(':')
-          next if has_ancestor?(node, %w(pre code tt))
+          next if has_ancestor?(node, ignored_ancestor_tags)
```

Comment:
I like this change, but I'd skip the `ignored_pattern`. This filter is designed to run on HTML fragments, so we should assume that tags is a reasonable input to work with.